Classes/Aspects/ThumbnailAspect.php

```diff
@@ -74,52 +76,24 @@ public function optimizeThumbnail(JoinPointInterface $joinPoint) {
 		$binaryRootPath = 'Private/Library/node_modules/';
 		$file = escapeshellarg($pathAndFilename);
 		$imageType = $thumbnailResource->getMediaType();
-		switch ($imageType) {
-			case 'image/jpeg':
-				if ($this->settings['formats']['jpg']['enabled'] === FALSE) {
-					return;
-				}
-				$library = 'jpegtran';
-				$binaryPath = sprintf('%1$s-bin/vendor/%s', $library);
-				$arguments = sprintf('-copy none -optimize %s -outfile %s %s', $this->settings['formats']['jpg']['progressive'] === TRUE ? '-progressive' : '', $file, $file);
-				if ($this->settings['formats']['jpg']['useGlobalBinary'] === TRUE) {
-					$useGlobalBinary = TRUE;
-				}
-				break;
-			case 'image/png':
-				if ($this->settings['formats']['png']['enabled'] === FALSE) {
-					return;
-				}
-				$library = 'optipng';
-				$binaryPath = sprintf('%1$s-bin/vendor/%s', $library);
-				$arguments = sprintf('-o%u -strip all -out %s %s', $this->settings['formats']['png']['optimizationLevel'], $file, $file);
-				if ($this->settings['formats']['png']['useGlobalBinary'] === TRUE) {
-					$useGlobalBinary = TRUE;
-				}
-				break;
-			case 'image/gif':
-				if ($this->settings['formats']['gif']['enabled'] === FALSE) {
-					return;
-				}
-				$library = 'gifsicle';
-				$binaryPath = sprintf('%1$s/vendor/%1$s', $library);
-				$arguments = sprintf('--batch -O%u %s ', $this->settings['formats']['gif']['optimizationLevel'], $file);
-				if ($this->settings['formats']['gif']['useGlobalBinary'] === TRUE) {
-					$useGlobalBinary = TRUE;
-				}
-				break;
-			case 'image/svg+xml':
-				if ($this->settings['formats']['svg']['enabled'] === FALSE) {
-					return;
-				}
-				$library = 'svgo';
-				$binaryPath = sprintf('%1$s/bin/%1$s', $library);
-				$arguments = sprintf('%s %s', $this->settings['formats']['svg']['pretty'] === TRUE ? '--pretty' : '', $file);
-				if ($this->settings['formats']['svg']['useGlobalBinary'] === TRUE) {
-					$useGlobalBinary = TRUE;
-				}
-				break;
+
+		if (!array_key_exists($imageType, $this->settings['formats'])) {
+			return;
+		}
+
+		$settings = $this->settings['formats'][$imageType];
+
+		$library = $settings['library'];
+		$binaryPath = $settings['binaryPath'];
+		$expression = $settings['arguments'];
+		$context = new ProtectedContext($settings);
+		$context->push($file, 'file');
```

Comment:
Currently the whole library configuration is put into the context, this way the previous settings for `pretty`, `progressive` and so on are still there. See line 89.
But I agree that it would be more intuitive to put it into an extra level. I'll change that.